planner/core/logical_plan_builder.go

```diff
@@ -903,6 +903,60 @@ func (b *PlanBuilder) coalesceCommonColumns(p *LogicalJoin, leftPlan, rightPlan
 		if err != nil {
 			return err
 		}
+	} else {
+		// Even with no using filter, we still should check the checkAmbiguous name before we try to find the common column from both side.
+		// (t3 cross join t4) natural join t1
+		//  t1 natural join (t3 cross join t4)
+		// t3 and t4 may generate the same name column from cross join.
+		// for every common column of natural join, the name from right or left should be exactly one.
+		checkAmbiguous := func(lnames, rnames types.NameSlice) error {
```

Comment:
I think we don't need to make it a separate function here because we don't reuse this logic.

Reply:
make sense